src/core/tracing.cc

```diff
@@ -28,87 +28,63 @@
-  }
-
-  if (trace_name.empty()) {
-    return Status(
-        RequestStatusCode::INVALID_ARG,
-        "trace configuration requires a non-empty trace name");
-  }
-
-  if (hostname.empty()) {
-    return Status(
-        RequestStatusCode::INVALID_ARG,
-        "trace configuration requires a non-empty host name");
-  }
-
-  if (port == 0) {
-    return Status(
-        RequestStatusCode::INVALID_ARG,
-        "trace configuration requires a non-zero port");
-  }
-
-  LOG_INFO << "Configured trace: " << trace_name << ", " << hostname << ":"
-           << port;
-
-  singleton_.reset(new TraceManager(trace_name, hostname, port));
-  return Status::Success;
+  activity_fn_(
+      reinterpret_cast<TRTSERVER_Trace*>(this), TRTSERVER_TRACE_REQUEST_START,
+      TIMESPEC_TO_NANOS(infer_stats->Timestamp(
+          ModelInferStats::TimestampKind::kRequestStart)),
+      activity_userp_);
+  activity_fn_(
+      reinterpret_cast<TRTSERVER_Trace*>(this), TRTSERVER_TRACE_QUEUE_START,
+      TIMESPEC_TO_NANOS(
+          infer_stats->Timestamp(ModelInferStats::TimestampKind::kQueueStart)),
+      activity_userp_);
+  activity_fn_(
+      reinterpret_cast<TRTSERVER_Trace*>(this), TRTSERVER_TRACE_COMPUTE_START,
+      TIMESPEC_TO_NANOS(infer_stats->Timestamp(
+          ModelInferStats::TimestampKind::kComputeStart)),
+      activity_userp_);
+  activity_fn_(
+      reinterpret_cast<TRTSERVER_Trace*>(this), TRTSERVER_TRACE_COMPUTE_END,
+      TIMESPEC_TO_NANOS(
+          infer_stats->Timestamp(ModelInferStats::TimestampKind::kComputeEnd)),
+      activity_userp_);
+  activity_fn_(
+      reinterpret_cast<TRTSERVER_Trace*>(this), TRTSERVER_TRACE_REQUEST_END,
+      TIMESPEC_TO_NANOS(
+          infer_stats->Timestamp(ModelInferStats::TimestampKind::kRequestEnd)),
+      activity_userp_);
 }
 
-TraceManager::TraceManager(
-    const std::string& trace_name, const std::string& hostname, uint32_t port)
-    : level_(0 /* disabled */), rate_(1000)
+#if 0
+void
+Log(const struct timespec& ts, const std::string& str)
 {
-  auto transportType = cppkin::TransportType::Http;
-  auto encodingType = cppkin::EncodingType::Json;
-
-  cppkin::CppkinParams cppkin_params;
-  cppkin_params.AddParam(cppkin::ConfigTags::HOST_ADDRESS, hostname);
-  cppkin_params.AddParam(cppkin::ConfigTags::PORT, port);
-  cppkin_params.AddParam(cppkin::ConfigTags::SERVICE_NAME, trace_name);
-  cppkin_params.AddParam(cppkin::ConfigTags::SAMPLE_COUNT, 1);
-  cppkin_params.AddParam(
-      cppkin::ConfigTags::TRANSPORT_TYPE,
-      cppkin::TransportType(transportType).ToString());
-  cppkin_params.AddParam(
-      cppkin::ConfigTags::ENCODING_TYPE,
-      cppkin::EncodingType(encodingType).ToString());
-
-  cppkin::Init(cppkin_params);
+  LOG_INFO << str << ts.tv_sec * NANOS_PER_SECOND + ts.tv_nsec;
 }
-
-Status
-TraceManager::SetLevel(uint32_t level, uint32_t rate)
+#endif
+#if 0
+void
+Trace::Submit(const std::vector<struct timespec>& timestamps)
```

Comment:
What will this two functions be used for? Logging to console in addition to writing to file?

Reply:
Those were left in by mistake. I will remove them in the next PR.